haystack/tools/from_function.py

```diff
@@ -3,16 +3,20 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import inspect
-from typing import Any, Callable, Dict, Optional
+from typing import Any, Callable, Dict, Optional, Union
 
 from pydantic import create_model
 
-from haystack.tools.errors import SchemaGenerationError
-from haystack.tools.tool import Tool
+from .errors import SchemaGenerationError
+from .tool import Tool
 
 
 def create_tool_from_function(
-    function: Callable, name: Optional[str] = None, description: Optional[str] = None
+    function: Callable,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    inputs: Optional[Dict[str, str]] = None,
+    outputs: Optional[Dict[str, Dict[str, Any]]] = None,
```

Comment:
to be consistent with `Tool` and `ComponentTool`, I would name these input parameters `inputs_from_state` and `outputs_from_state`. WDYT?

Reply:
Very good point! Yes I will do it now and then merge. Better rename now than having a breaking change later.